src/screens/ChannelListScreenV2/index.tsx

```diff
@@ -1,166 +1,72 @@
+import PostNotificationChannelItem from '../../components/AnonymousChat/PostNotificationChannelItem';
+import Search from '../ChannelListScreen/elements/Search';
+import useAnonymousChannelListScreenHook from '../../hooks/screen/useAnonymousChannelListHook';
 import useLocalDatabaseHook from '../../database/hooks/useLocalDatabaseHook';
-import useUserAuthHook from '../../hooks/core/auth/useUserAuthHook';
 import useRootChannelListHook from '../../hooks/screen/useRootChannelListHook';
-import {fcmTokenService} from '../../service/users';
-import StorageUtils from '../../utils/storage';
-import ChannelListScreen from '../ChannelListScreen';
-import Search from '../ChannelListScreen/elements/Search';
-import AnonymousChannelListScreen from './AnonymousChannelListScreen';
-import {
-  PERMISSION_STATUS_ACCEPTED,
-  PERMISSION_STATUS_BLOCKED,
-  PERMISSION_STATUS_PENDING
-} from '../../utils/constants';
 
-const ChannelListScreenV2 = () => {
+const ChannelListScreenV2 = ({route}) => {
   const {refresh} = useLocalDatabaseHook();
+  const {checkNotificationPermission} = useRootChannelListHook();
   const navigation = useNavigation();
-  const {profile} = useUserAuthHook();
   const isFocused = navigation.isFocused();
-
-  const [selectedTab, setSelectedTab] = React.useState(0);
-  const {signedChannelUnreadCount, anonymousChannelUnreadCount} = useRootChannelListHook();
-
-  const navigateToContactScreen = () => {
-    navigation.navigate('ContactScreen');
-  };
-
-  const requestPermission = async () => {
-    const authStatus = await messaging().requestPermission();
-    const enabled =
-      authStatus === messaging.AuthorizationStatus.AUTHORIZED ||
-      authStatus === messaging.AuthorizationStatus.PROVISIONAL;
-
-    if (enabled) {
-      const fcmToken = await messaging().getToken();
-      const payload = {
-        fcm_token: fcmToken
-      };
-      fcmTokenService(payload);
-    }
-  };
+  const {
+    channels: anonChannels,
+    goToChatScreen,
+    goToPostDetailScreen,
+    goToContactScreen
+  } = useAnonymousChannelListScreenHook();
 
   React.useEffect(() => {
-    const checkNotif = async () => {
-      if (!navigation.isFocused()) return;
-      PushNotification.checkPermissions(
-        (
-          data: PushNotificationPermissions & {
-            authorizationStatus?: number;
-          }
-        ) => {
-          const lastPromptTime = StorageUtils.lastPromptNotification.get();
-          const promptInterval = 48 * 60 * 60 * 1000;
-          const currentTime = new Date().getTime();
-          if (lastPromptTime && currentTime - parseFloat(lastPromptTime) < promptInterval) {
-            return; // Don't show the prompt if the interval has not passed
-          }
-
-          const needsPermission =
-            data.alert === false || data.badge === false || data.sound === false;
+    if (isFocused) {
+      refresh('channelList');
+      checkNotificationPermission();
+    }
+  }, [isFocused]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `useEffect` hook is used to refresh the channel list and check notification permissions when the screen is focused. However, it's important to note that the `isFocused` value from `useNavigation` may not always update as expected, which could lead to unexpected behavior. It would be better to use the `useFocusEffect` hook from `@react-navigation/native` instead, which ensures the effect runs when the screen comes into focus and cleans up when it goes out of focus.

Here's how you can refactor this:

```diff
- React.useEffect(() => {
-   if (isFocused) {
-     refresh('channelList');
-     checkNotificationPermission();
-   }
- }, [isFocused]);
+ import { useFocusEffect } from '@react-navigation/native';
+
+ useFocusEffect(
+   React.useCallback(() => {
+     refresh('channelList');
+     checkNotificationPermission();
+   }, [])
+ );
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->